### PR TITLE
docs: update environ var

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
- [x] Format this section
- [x] Rearrange the order
- [x] Add: JINA_CONTRIB_MODULE_IS_LOADING, JINA_LOG_CONFIG, JINA_LOG_WORKSPACE, JINA_RANDOM_PORTS, JINA_RANDOM_PORT_MAX, JINA_RANDOM_PORT_MIN
- [x] Delete legacy var

### CODE REVIEW

File: chapters/envs.rst [hunk 1]
```diff
@@ -1,8 +1,10 @@
+=====================================
 OS Environment Variables Used in Jina
 =====================================
 
-
-
+.. meta::
+   :description: A list of environment variables that ``jina`` respects during runtime.
```
Comment (JoanFM) [inline chapters/envs.rst]: jina respects? should it be uses?

File: chapters/envs.rst [hunk 2]
```diff
@@ -14,157 +16,168 @@ Here is the list of environment variables that ``jina`` respects during runtime.
...
     :default: unset
 
-.. confval:: JINA_PROFILING
-
-    Set to any non-empty value to turn on service-level time profiling for Jina.
-
-    :default: unset
+.. confval:: JINA_BINARY_DELIMITER
 
+    The delimiter used in :class:`BasePbIndexer`. Often use a delimiter phrase that you can determine does not occur. (Think of the mime message boundaries)
```
Comment (JoanFM) [inline chapters/envs.rst:32]: what are the mime message boundaries

Comment (JoanFM) [reply chapters/envs.rst:32]: that does not occur in any of the messages

File: chapters/envs.rst [hunk 3]
```diff
@@ -14,157 +16,168 @@ Here is the list of environment variables that ``jina`` respects during runtime.
...
-    Paths of the third party components.
+    :default: `0.0.0.0`
+
+.. confval:: JINA_DISABLE_UVLOOP
+
+    ``uvloop`` is ultra fast implementation of the asyncio event loop on top of libuv. Since Jina 0.3.6, Jina relies on ``uvloop`` to manage the sockets and streams.
+
+    :default: unset
+
+.. confval:: JINA_EXECUTOR_WORKDIR
```
Comment (JoanFM) [inline chapters/envs.rst:64]: is this being used?

File: chapters/envs.rst [hunk 4]
```diff
@@ -14,157 +16,168 @@ Here is the list of environment variables that ``jina`` respects during runtime.
...
 
 .. confval:: JINA_IPC_SOCK_TMP
 
     Temp directory when using IPC sockets for the control port, not used on Windows system or when the control port is over TCP sockets.
 
-.. confval:: JINA_LOG_FILE
+.. confval:: JINA_LOG_CONFIG
 
-    Control where the logs output to. Possible values: ``TXT`` for plain text format, which is the uncolored version of the screen output. ```JSON`` for structured log output.
+    The configuration file for the logger. If unset then will load from logging.default.yml.
```
Comment (JoanFM) [inline chapters/envs.rst]: under resources

File: chapters/envs.rst [hunk 5]
```diff
@@ -14,157 +16,168 @@ Here is the list of environment variables that ``jina`` respects during runtime.
...
 
-    :default: unset, aka stdout only.
+    :default: unset
 
-.. confval:: JINA_SOCKET_HWM
+.. confval:: JINA_LOG_ID
```
Comment (JoanFM) [inline chapters/envs.rst:86]: is this still used?

File: chapters/envs.rst [hunk 6]
```diff
@@ -14,157 +16,168 @@ Here is the list of environment variables that ``jina`` respects during runtime.
...
+    :default: ``INFO``
 
 .. confval:: JINA_LOG_NO_COLOR
 
     Show colored logs in stdout, set to any non-empty value to disable the color log, e.g. if you want to pipe the log into other apps.
 
     :default: unset
 
-.. confval:: JINA_EXECUTOR_WORKDIR
+.. confval:: JINA_LOG_WORKSPACE
```
Comment (JoanFM) [inline chapters/envs.rst:105]: where is this used?

File: chapters/envs.rst [hunk 7]
```diff
@@ -14,157 +16,168 @@ Here is the list of environment variables that ``jina`` respects during runtime.
...
 
-    ``uvloop`` is ultra fast implementation of the asyncio event loop on top of libuv. Since Jina 0.3.6, Jina relies on ``uvloop`` to manage the sockets and streams.
+.. confval:: JINA_WARN_UNNAMED
+
+    Set to any non-empty value to turn on the warning for unnamed executors.
 
     :default: unset
 
-.. confval:: JINA_RAISE_ERROR_EARLY
+.. confval:: JINA_WORKSPACE
```
Comment (JoanFM) [inline chapters/envs.rst:171]: where is this used?

File: chapters/envs.rst [hunk 8]
```diff
@@ -14,157 +16,168 @@ Here is the list of environment variables that ``jina`` respects during runtime.
...
     :default: unset
 
-.. confval:: JINA_LOG_ID
 
-    The identifier of a logger so that it can be used as group identifier by ``fluentd``. It is set when Pea starts a new process
-    to allow grouping by pod identifier for pea and executor logger.
 
-    :default: unset
\ No newline at end of file
+
```
Comment (JoanFM) [inline chapters/envs.rst]: please remove extra lines

**Review by JoanFM — CHANGES_REQUESTED** [review]
Are we sure all of these env vars are actually used in the core?

**Review by FionnD — APPROVED** [review]

File: chapters/envs.rst (hunk 3 above)
Comment (Yongxuanzhang) [reply chapters/envs.rst:64]: I think so. I checked with the list in jina/init.py. And this is used in jina/executor/init.py

File: chapters/envs.rst (hunk 2 above)
Comment (Yongxuanzhang) [reply chapters/envs.rst:32]: This is from this previous doc I didn't make any change. So to my understanding, this is just an example to explain delimiter, but this example may not be so good?

File: chapters/envs.rst (hunk 1 above)
Comment (Yongxuanzhang) [reply chapters/envs.rst]: ✅

File: chapters/envs.rst (hunk 4 above)
Comment (Yongxuanzhang) [reply chapters/envs.rst]: ✅

File: chapters/envs.rst (hunk 5 above)
Comment (Yongxuanzhang) [reply chapters/envs.rst:86]: This JINA_LOG_ID is used in Logger

File: chapters/envs.rst (hunk 6 above)
Comment (Yongxuanzhang) [reply chapters/envs.rst:105]: JINA_LOG_WORKSPACE  is used in logging/logger.py

File: chapters/envs.rst (hunk 7 above)
Comment (Yongxuanzhang) [reply chapters/envs.rst:171]: JINA_WORKSPACE  is used in optimizer/discovery.py and mostly used in tests, examples.

Comment (JoanFM) [reply chapters/envs.rst:171]: if is used by discovery is good, but in tests and examples is not cores concern, here there should only be listed those used in the core

File: chapters/envs.rst (hunk 2 above)
Comment (JoanFM) [reply chapters/envs.rst:32]: not sure

Comment (Yongxuanzhang) [reply chapters/envs.rst:32]: How about adding (Think of the mime message boundaries. e.g. B164240059B29C0E4EFEC397)

File: chapters/envs.rst (hunk 8 above)
Comment (Yongxuanzhang) [reply chapters/envs.rst]: Done

File: chapters/envs.rst (hunk 7 above)
Comment (JoanFM) [reply chapters/envs.rst:171]: Change the description to the one in `discovery`.

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: chapters/envs.rst (hunk 7 above)
Comment (Yongxuanzhang) [reply chapters/envs.rst:171]: Just changed

**Review by JoanFM — APPROVED** [review]

**Pull request merged**